### PR TITLE
Add support for Investigator's versatile vials

### PR DESCRIPTION
**Pull request by ebuckle**
Resolves https://github.com/foundryvtt/pf2e/issues/2450

image

Add tracking for versatile vials resource with REs on relevant investigator features, as well as a toggle for the Quick Tincture ability, simiar to existing Quick Alchemy.

### CODE REVIEW

**Comment by ebuckle** [comment]
Just pending a fix to the migration, probably tomorrow since it's late

**Comment by ebuckle** [comment]
All good 👍

File: src/module/actor/character/data/sheet.ts [hunk 1]
```diff
@@ -34,18 +34,25 @@ export interface CraftingEntriesSheetData {
...
+    resources: {
+        infusedReagents: {
+            value: number;
+            max: number;
+        };
+        versatileVials: {
+            value: number;
+            max: number;
+        };
+    };
```
Comment (stwlam) [inline src/module/actor/character/data/sheet.ts:55]: This is isn't going to work. We're not hard-coding every advanced-alchemy-like feature ever published and to be published.

Comment (ebuckle) [reply src/module/actor/character/data/sheet.ts]: Attempt made to make this more generic.

Should I also extend this to crafting resources not being strictly typed?

**Pull request closed without merging**